### ISSUE TITLE
FR: A hint for books in collections

### ISSUE THREAD

**Issue by ichnilatis-gr** [koreader/koreader#11846]
This issues has been discussed [here](https://github.com/koreader/koreader/issues/6054). 

Now that the [collection feature](https://github.com/koreader/koreader/pull/11693) has been added (thank you for this), there is the following "problem": Even if a book has been added to a collection, in the pop-up window, after long pressing on a book, there is the option "add to collection". So we can't know, even with this practical way, if the book has been added to a collection. I mean, when only Favorites existed, this was a way to see it, as, in case that the book had been added to Favorites, there was the option "Remove from Favorites" (something like that).
So, now there is need for a hint (both for list and mosaic mode) that will indicate if a book has been added to a collection.
@hius07 I remind your [promise](https://github.com/koreader/koreader/issues/6054#issuecomment-1818739293) 🙂 

Thank you!

**Comment by hius07** [comment]
For Classic and List mode it can be an empty star in the mandatory.
Grid mode requires more design thinking.

![1](https://github.com/koreader/koreader/assets/62179190/0491752c-2412-4a81-84a3-4edda3d9eab3)

![2](https://github.com/koreader/koreader/assets/62179190/0c9cea72-47e0-49db-a679-d99c79d3eeea)

![3](https://github.com/koreader/koreader/assets/62179190/86131398-8a6f-4859-ae55-ad2a98fa6b80)

![4](https://github.com/koreader/koreader/assets/62179190/2f99bfa8-6035-4576-97ce-3d3efda3ab34)

**Comment by hius07** [comment]
In Grid text cover the star can be appended at the beginning of an author, a title, or a filename.
(I added the folder to folder shortcuts for comparison)

![5](https://github.com/koreader/koreader/assets/62179190/07f316ea-1037-4dcc-b220-240b393023b0)

No good idea for image cover so far.

![6](https://github.com/koreader/koreader/assets/62179190/522b48c9-50ba-4884-a2f0-eb61caf3ed7f)

**Comment by ichnilatis-gr** [comment]
I give you an example from Pocketbook. They use a heart inside a white circle. We could use a star at the upper right corner of the book's cover.

![scr0009 (Medium)](https://github.com/koreader/koreader/assets/76203123/c66657cb-f1a9-435b-9361-caa60869ba8b)

**Comment by hius07** [comment]
To remain in consistency with our corner mark it can be:

![1](https://github.com/koreader/koreader/assets/62179190/1db65d33-3020-4ca5-b0af-52fdff72eb05)

![2](https://github.com/koreader/koreader/assets/62179190/46ea7506-0a33-45bd-a5b9-bd9fa4462c33)

![3](https://github.com/koreader/koreader/assets/62179190/2ba1dcf6-545d-4731-a1dd-c75b8ced6862)

![4](https://github.com/koreader/koreader/assets/62179190/92b4a914-2688-44ca-8e2c-c927adc23ce4)

**Comment by Frenzie** [comment]
I think that was just meant to show the concept, not a lack of transparency. Unless I'm misreading it. ^_^

**Comment by hius07** [comment]
> transparent star

![1](https://github.com/koreader/koreader/assets/62179190/e5e0aee7-e131-4af2-91bf-ac2d391128b0)

**Comment by ichnilatis-gr** [comment]
It is indeed indistinct.

**Comment by hius07** [comment]
> white star

![1](https://github.com/koreader/koreader/assets/62179190/b994bd11-5b71-4127-8e8b-1562f0c2d4dc)

**Comment by ichnilatis-gr** [comment]
I have to think before I speak... 🙂

**Comment by poire-z** [comment]
Could it be a white-filled star with thick-black-borders and a transparent outside ?
Or, as does Pocketbook, a (solid filled or not) black star inside a solide white disc. It may look better (can't be sure before seeing it :)) than the solid white square ?

**Comment by ichnilatis-gr** [comment]
> Could it be a white-filled star with thick-black-borders and a transparent outside ?

I think that's what I had in mind...

**Comment by Frenzie** [comment]
> I think that's what I had in mind...

Same, yes. ;-)

**Comment by hius07** [comment]
![1](https://github.com/koreader/koreader/assets/62179190/9a3addd7-44e3-40c3-a1f5-86ec4af0711e)

![2](https://github.com/koreader/koreader/assets/62179190/62535ab6-a664-4140-8806-0cc66935f2de)

![3](https://github.com/koreader/koreader/assets/62179190/e177ec60-06a6-40e0-a6f4-b7324ef163b0)

**Comment by hius07** [comment]
Smaller star

![7](https://github.com/koreader/koreader/assets/62179190/2b9f7bc8-6494-4ac3-b52f-f19e09d60bbd)

![8](https://github.com/koreader/koreader/assets/62179190/f2e1ccaa-9fc8-4c79-b664-4cf2b7395131)

**Comment by hius07** [comment]
> a white-filled star with thick-black-borders and a transparent outside ?

![1](https://github.com/koreader/koreader/assets/62179190/c3ccc6b9-ec90-48fb-b649-590e30448bed)

![2](https://github.com/koreader/koreader/assets/62179190/7db61c48-c87f-499d-b949-a7c375c1086e)

![3](https://github.com/koreader/koreader/assets/62179190/b702ca24-eb92-4d74-b025-fffd82e057f6)

**Comment by Frenzie** [comment]
With the white circle around it is probably clearer but either way seems okay.

**Comment by ichnilatis-gr** [comment]
I would prefer the white-filled star with thick-black-borders and a transparent outside. The other one with the circle I think looks like an emblem.

**Comment by hius07** [comment]
Final comparison:

![1](https://github.com/koreader/koreader/assets/62179190/5aba4edb-2dc7-4425-ba43-75293f7d2da4)

![2](https://github.com/koreader/koreader/assets/62179190/1312edab-3a26-4321-97d8-7f6e903470f7)

**Comment by benoit-pierre** [comment]
Second one, for me.

**Comment by ichnilatis-gr** [comment]
If the star with the circle will be preferred, I think the black star with the white circle would be better.

**Comment by Frenzie** [comment]
The reason I think with the circle is slightly better is because otherwise it somewhat disappears into darker backgrounds. Perhaps one with an additional thin white line around it to see what that would look like?

**Comment by hius07** [comment]
> the black star with the white circle would be better

Added to the above post with screenshots.

**Comment by ichnilatis-gr** [comment]
> Added to the above post with screenshots.

Yes, I've seen it. I don't mean to create it.

**Comment by hius07** [comment]
> Perhaps one with an additional thin white line around it to see what that would look like?

Thin line - white circle, or thin line - white star outlining the black star?

**Comment by jonnyl2** [comment]
Of the above options, my preference would be the 2nd one (star with no background). (Possibly the star could be enlarged a bit if it seems not discernible enough for some.)

If a background, then I think a square background in the top-right corner (maybe with something like 50% opacity) would look better than a circle, as it would blend better with the rectangle-shaped cover. Star on circle on rectangle seems like a bit too many shapes in one per cover.

**Comment by Frenzie** [comment]
> Thin line - white circle, or thin line - white star outlining the black border of the star?

The latter.

**Comment by hius07** [comment]
> > Thin line - white circle, or thin line - white star outlining the black border of the star?
> 
> The latter.

We need an artist to draw the new svg icon for that.

**Comment by hius07** [comment]
I personally tend to option 2.
Black star with white filling is visible well on dark and light covers.

**Comment by Frenzie** [comment]
Option 2 is fine.

**Comment by hius07** [comment]
Should we show a star when are in a collection (all books would have the mark, obviously).

**Comment by poire-z** [comment]
I don't know if there exists a solution to this, but with the standalone star, it feels it is larger on a white bg cover, and it feels quite smaller on a dark bg cover:
![image](https://github.com/koreader/koreader/assets/24273478/b715fe35-8ed8-4a0a-976c-fed2fd00c433)

**Comment by Frenzie** [comment]
> Should we show a star when browsing a collection (all books would have the mark, obviously)?

Seems unnecessary.

**Comment by poire-z** [comment]
The white circle is probably not the solution :) I get the same feeling with your circle screenshots.
Maybe @offset-torque (with his background in human perception and studies) has an idea ? (without having to guess the "tone" (light/dark) of the cover).

**Comment by offset-torque** [comment]
> **poire-z** - Could it be a white-filled star with thick-black-borders and a transparent outside ?

This is the correct solution. Also you see this in movie subtitles where background is unpredictable. A thick dark border with bright text has the best visibility.

> **poire-z** - I don't know if there exists a solution to this, but with the standalone star, it feels it is larger on a white bg cover, and it feels quite smaller on a dark bg cover

Unfortunately this is the side effect, border joins either to foreground or background.

Best solution is to provide a stable bg color as **jonnyl2** said:

> **jonnyl2** - If a background, then I think a square background in the top-right corner (maybe with something like 50% opacity)

A dark square or a dogear to cover the star's background. In this case star can be just white too, since border wouldn't be necessary.

Also I agree that star should be enlarged a bit as **jonnyl2** said.


A mockup (quickly hand drawn, not to scale):
![dogear_collection](https://github.com/koreader/koreader/assets/84420582/b1ce47b4-f6fd-4848-bc91-15e506f49d81)

**Comment by hius07** [comment]
> A dark square or a dogear to cover the star's background.

Can it be the square?
We have a triangle at the bottom (book status hint), if a dogear is of the same size, the star will be too small.

![1](https://github.com/koreader/koreader/assets/62179190/5b95e10f-b50f-4581-bc2f-6ae7236b2bea)

**Comment by offset-torque** [comment]
Why not. Main idea is giving it a stable background so users can immediately recognize it without the book cover interfering with the outline of the star.

Can we combine these two indicators, something like:
- "Empty triangle" for book status
-  "Filled triangle with star" for collection

Since lower right corner is the least important part of the book cover (no title there, generally empty) we can use that corner for both indicators.

**Comment by hius07** [comment]
Book on hold

![3](https://github.com/koreader/koreader/assets/62179190/f0fa9430-279a-4014-bef3-fda474adcf16)

**Comment by offset-torque** [comment]
Star in this last screenshot looks pleasant and easily recognizable to me.

**Comment by jonnyl2** [comment]
Can the square be made partially transparent to let some of the cover image shine through?

**Comment by offset-torque** [comment]
> Can the square be made partially transparent to let some of the cover image shine through?

This will probably interfere with the instant shape recognition of the star and will decrease the icon/background contrast. My perspective is that since we already ruin that corner of the cover, at least we can try to maximize ux effectiveness. 

I want to remind everyone that everything looks much nicer in screenshots with their perfect contrast and lighting until they meet the reality of our e-ink screens :)

**Comment by Frenzie** [comment]
> I want to remind everyone that everything looks much nicer in screenshots with their perfect contrast and lighting until they meet the reality of our e-ink screens :)

Largely disagree (well, not for partial transparency stuff), most things in that screenshot look much nicer on a real device. :-D

**Comment by offset-torque** [comment]
After **Frenzie**'s comment I got curious about **"why e-ink looks nicer?"** and found this relatively old study. Although they didn't specifically studied e-ink, I found their results fascinating and wanted to share this useful information with my fellow e-ink enthusiasts.

According to their findings, a **recent generation e-ink screen in an office with standard lighting** (this is approximate because I calculated/converted) has the optimal contrast ratio for comfortable reading. I guess this explains why these screens are less tiring (which I was always curious about). 

Probably colors on other screen types like LCD can be adjusted for a similar improvement but the advantage of e-ink comes from its nature. Apparently when we put pure black text on pure white background on an e-ink screen, we automatically reach this optimum reading contrast because the substrate softens both ends (black and white) acting as a natural clamp filter.

As a speculation, I think this means probably we reached the peak with the current gen grayscale screens and all the improvements from now on will come from the color e-ink technology.

Here is the link to the study if you want to read the details:
[Effects of polarity and luminance contrast on visual performance and VDT display quality](https://www.sciencedirect.com/science/article/abs/pii/S0169814199000402)
